shared/src/models/patient-health-state.ts

```diff
@@ -62,54 +21,86 @@ export class ConditionParameters {
      */
     @IsOptional()
     @IsNumber()
+    @Min(0)
     public readonly earliestTime?: number;
+
     @IsOptional()
     @IsNumber()
+    @Min(0)
     public readonly latestTime?: number;
-    @IsOptional()
-    @IsValidHealthPoint()
-    public readonly minimumHealth?: HealthPoints;
-    @IsOptional()
-    @IsValidHealthPoint()
-    public readonly maximumHealth?: HealthPoints;
+
     @IsOptional()
     @IsBoolean()
     public readonly isBeingTreated?: boolean;
+
+    @IsOptional()
+    @IsNumber()
+    @Min(0)
+    public readonly requiredMaterialAmount?: number;
+
+    @IsOptional()
+    @IsNumber()
+    @Min(0)
+    public readonly requiredNotArztAmount?: number;
+
+    @IsOptional()
+    @IsNumber()
+    @Min(0)
+    public readonly requiredNotSanAmount?: number;
+
+    @IsOptional()
+    @IsNumber()
+    @Min(0)
+    public readonly requiredRettSanAmount?: number;
+
+    @IsOptional()
+    @IsNumber()
+    @Min(0)
+    public readonly requiredSanAmount?: number;
     /**
      * The id of the patients healthState to switch to when all the conditions match
      */
-    @IsUUID(4, uuidValidationOptions)
-    public readonly matchingHealthStateId: UUID;
+    @IsString()
+    public readonly matchingHealthStateId: string;
 
     /**
      * @deprecated Use {@link create} instead
      */
     constructor(
         earliestTime: number | undefined,
         latestTime: number | undefined,
-        minimumHealth: HealthPoints | undefined,
-        maximumHealth: HealthPoints | undefined,
         isBeingTreated: boolean | undefined,
+        requiredMaterialAmount: number | undefined,
+        requiredNotArztAmount: number | undefined,
+        requiredNotSanAmount: number | undefined,
+        requiredRettSanAmount: number | undefined,
+        requiredSanAmount: number | undefined,
         matchingHealthStateId: UUID
     ) {
         this.earliestTime = earliestTime;
         this.latestTime = latestTime;
-        this.minimumHealth = minimumHealth;
-        this.maximumHealth = maximumHealth;
         this.isBeingTreated = isBeingTreated;
+        this.requiredMaterialAmount = requiredMaterialAmount;
+        this.requiredNotArztAmount = requiredNotArztAmount;
+        this.requiredNotSanAmount = requiredNotSanAmount;
+        this.requiredRettSanAmount = requiredRettSanAmount;
+        this.requiredSanAmount = requiredSanAmount;
         this.matchingHealthStateId = matchingHealthStateId;
     }
 
     static readonly create = getCreate(this);
 }
 
 export class PatientHealthState {
-    @IsUUID(4, uuidValidationOptions)
-    public readonly id: UUID = uuid();
+    @IsString()
+    public readonly id: string;
```

Comment:
Rename to `name`. This should also include all variables and properties named after this field, but I'd be fine with letting them as they are for sake of the remaining time.